### ISSUE TITLE
Optimized bundle build doesn't respect `rego.v1` and `future.keywords` imports

### ISSUE THREAD

**Issue by johanfylling** [open-policy-agent/opa#6450]
When building a bundle from the following module with `opa build -O1`:

```rego
package test

import rego.v1

# METADATA
# entrypoint: true
p if {
    input.x == 1
}
```

the resulting optimized support module will drop the `rego.v1` import and `if` keyword:

```rego
package test

p {
    input.x = 1
}
```

The same issue occurs when the original module imports `future.keywords` instead of `rego.v1`.

**Comment by xico42** [comment]
@johanfylling regarding this issue, I've come to realize that the SDK during bundle plugin activation does not respect the V1 compatibility flag provided.

This is because the `bundle.ActivateOpts` struct [is instantiated](https://github.com/open-policy-agent/opa/blob/main/plugins/bundle/plugin.go#L599-L607) without forwarding the plugin manager parser options provided during SDK initialization.

**Comment by xico42** [comment]
I am not totally sure that the problem I faced is really related to this issue. But anyway, I have provided a failing test case with the fix at #6689.

Also, it seems to only happen when there are multiple bundles being loaded. Also, by taking a look at the logs it seems that OPA is mixing up the bundles together, because the syntax error logged is for a source code of a bundle other than the one the status is logged for.
